compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -147,52 +143,49 @@ pub fn intern_const_alloc_recursive<
     // raw pointers, so we cannot rely on validation to catch them -- and since interning runs
     // before validation, and interning doesn't know the type of anything, this means we can't show
     // better errors. Maybe we should consider doing validation before interning in the future.
-    while let Some((alloc_id, mutability)) = todo.pop() {
+    while let Some(prov) = todo.pop() {
+        let alloc_id = prov.alloc_id();
         if ecx.tcx.try_get_global_alloc(alloc_id).is_some() {
             // Already interned.
             debug_assert!(!ecx.memory.alloc_map.contains_key(&alloc_id));
             continue;
         }
         just_interned.insert(alloc_id);
```

Comment:
Please move this next to `intern_shallow`, they conceptually belong together.

Reply:
uh, I forgot to remove this. it's dead code now anyway. We only needed it because we were handling interning separately from walking the nested allocs

Reply:
No, we need it to distinguish "pointers to memory that existed before this" from "pointers to newly interned memory". I think it is not dead at all.